packages/opentelemetry-plugin-http/src/utils.ts

```diff
@@ -217,6 +217,46 @@ export const setSpanWithError = (
   span.setStatus(status);
 };
 
+/**
+ * Adds attributes for content-length and content-encoding HTTP headers
+ * @param { OutgoingHttpHeaders | IncomingHttpHeaders } headers http headers
+ * @param { Attributes } Attributes span attributes
+ * @param { boolean } isRequest set true for setting request content-header
+ */
+export const setContentLengthAttributes = (
+  headers: OutgoingHttpHeaders | IncomingHttpHeaders,
+  attributes: Attributes,
+  isRequest: boolean
+) => {
+  let isCompressed = false;
+
+  if (headers['content-length'] === undefined) return;
+
+  if (
+    headers['content-encoding'] &&
+    headers['content-encoding'] !== 'identity'
+  ) {
+    isCompressed = true;
+  }
+
+  let key = null;
```

Comment:
In general I find this pattern off-putting:

```
let somevar = null;
if() {
  if() {
   somevar = ....
  }
  if() {
    somevar = ...
  }
  somevar = ...
}
```

I think you are trying to fit too much into a single function. Rather see it decomposed into `setRequestContentLengthAttribute` and `setResponseContentLengthAttribute`.

```
function setRequestContentLengthAttribute(
    request: IncomingMessage,
    attributes: Attributes,
) {
	const length = getContentLength(response)
    if (typeof length !== "number") return;

    if (isCompressed(request)) {
        attributes[HttpAttribute.HTTP_REQUEST_CONTENT_LENGTH] = length;
    } else {
        attributes[HttpAttribute. HTTP_REQUEST_CONTENT_LENGTH_UNCOMPRESSED] = length;
    }
}

function setResponseContentLengthAttribute(
    response: IncomingMessage,
    attributes: Attributes,
) {
	const length = getContentLength(response)
    if (typeof length !== "number") return;

    if (isCompressed(response)) {
        attributes[HttpAttribute.HTTP_RESPONSE_CONTENT_LENGTH] = length;
    } else {
        attributes[HttpAttribute. HTTP_RESPONSE_CONTENT_LENGTH_UNCOMPRESSED] = length;
    }
}

function getContentLength(message: IncomingMessage): number | null {
    const contentLengthHeader = message.getHeader('content-length');
	if (typeof contentLengthHeader !== "string" || isNaN(contentLengthHeader)) return null;
    return parseInt(contentLengthHeader);
}

function isCompressed(message: IncomingMessage) {
    const encoding = message.getHeader('content-encoding')
	return !!encoding && encoding !== 'identity';
}

```

Reply:
Good call, I'll push up a commit that breaks them out

Reply:
@carlo-808 and I paired on a commit that we just pushed. It splits this functions out.